src/storage/pooled_storage_manager.h

```diff
@@ -68,9 +69,12 @@ class GPUPooledStorageManager final : public StorageManager {
     DirectFreeNoLock(handle);
   }
 
+ private:
+  MemoryManager mm;
```

Comment:
Please use singleton design pattern. You can check Swap implementation for the detail.